bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -959,19 +979,31 @@ public void run() {
                             }
                             journalFlushWatcher.reset().start();
                             bc.flush(false);
+
+                            for (int i = 0; i < toFlush.size(); i++) {
```

Comment:
also I think I had a comment before : it's better not to put this loop in between
line 981: `bc.flush(false);` and
line 992: `lastFlushPosition = bc.position();`.

because there is a timer `journalFlushWatcher` that calculating the time spending on flushing. consider moving down to just before `if (syncData) {`. so all the stats/timer are correct as expected and easier to understand what metrics are measuring what logic.

Reply:
oaky, let me try to revert and do another proposal

Reply:
@sijie I have reverted the change and move the logic to a different branch. Tests are passing.
Is this what you are thinking ?

I wonder if this approach really reduces the number of fsyncs as anyway we are issueing a "ForceWriteRequest".
I would like that in the case all the entries written to the bookie are DEREFFED_SYNC the behaviour should be like having syncData = false

Reply:
@eolivelli 

This is different from using 'syncData = false'. because the goal here is add entries can complete before fsync, while the whole assumption of lac still remain. the `syncData = fase` is disabling fsync completely. These two cases have different durability guarantees.

Also I don't think the goal is to reduce the number of fsyncs. But if you want to reduce the number of fsyncs, you can anyway increase the flush commit interval or flush buffer size.